### PR TITLE
Fix for #4, rejoin is broken

### PR DESCRIPTION
**Pull request by chuckhays**
This is the same code that was in https://github.com/webrtc/samples/pull/462, but moved to the new repo and added a unit test.

@jiayliu @juberti

### CODE REVIEW

File: src/web_app/js/call_test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/* globals TestCase, SignalingChannel:true, requestUserMedia:true,
+   assertEquals, assertTrue */
+
+'use strict';
+
+var MockSignalingChannel = function() {
+};
+
+MockSignalingChannel.prototype.open = function() {
+  return new Promise(function(resolve) { resolve(); });
```
Comment (jiayliu) [inline src/web_app/js/call_test.js]: nit: just return Promise.resolve()?

File: src/web_app/js/call_test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  SignalingChannel = this.signalingChannelBackup_;
+  requestUserMedia = this.requestUserMediaBackup_;
+};
+
+CallTest.prototype.testRestartInitializesMedia = function() {
+  var call = new Call(this.params_);
+  var mediaStarted = false;
+  call.onlocalstreamadded = function(stream) {
+    mediaStarted = true;
+    assertEquals('stream', stream);
```
Comment (jiayliu) [inline src/web_app/js/call_test.js]: nit: move 'stream' to a const var

File: src/web_app/js/call.js [hunk 3]
```diff
@@ -40,6 +40,12 @@ var Call = function(params) {
   this.onsignalingstatechange = null;
   this.onstatusmessage = null;
 
+  this.getMediaPromise_ = null;
+  this.getTurnServersPromise_ = null;
+  this.initPromises_();
```
Comment (jiayliu) [inline src/web_app/js/call.js]: nit: a more descriptive name than initPromise_, e.g. requestMediaAndTurnServers.

File: src/web_app/js/call_test.js (hunk 1 above)
Comment (chuckhays) [inline src/web_app/js/call_test.js]: Sure, I had that originally, I think I changed it while trying to debug the Promise.all issue.

**Comment by chuckhays** [comment]
Updated. PTAL.

**Comment by jiayliu** [comment]
lgtm

**Pull request merged**